tests/target/issue-3170.rs

```diff
@@ -0,0 +1,38 @@
+fn main() {
+    let n = 1;
+    match n {
+        1 => {
+            // comment 1.1
+            // comment 1.2
+            1
+        }
+        2 =>
+        // comment 2.1
+        // comment 2.2
+        {
+            // comment 2.3
+            // comment 2.4
+            2
+        }
+        3 =>
+        // comment 3.1
+        // comment 3.2
+        {
+            3
+        }
+        4 => {
+            // comment 4.1
+            // comment 4.2
+            4
+        }
+        5 => 5,
+        6 => { // comment 6
+        }
```

Comment:
I did have to update this one line to get the tests to pass as the comment is now being kept on the same line as the opening bracket. IIRC that's because been some subequent updates that maintain that same line association. I added another arm below that I believe covers that original use case